src/openswathalgo/include/OpenMS/OPENSWATHALGO/ALGO/Scoring.h

```diff
@@ -131,10 +135,13 @@ namespace OpenSwath
     OPENSWATHALGO_DLLAPI void normalize_sum(double x[], unsigned int n);
 
     // Compute rank of vector elements
-    OPENSWATHALGO_DLLAPI std::vector<unsigned int> computeRank(const std::vector<double>& w);
+    OPENSWATHALGO_DLLAPI void computeRank(const std::vector<double>& v, std::vector<unsigned int>& ranks);
+
+    // Compute rank of vector elements for each row in a 2D array
```

Comment:
please make this a proper doxygen docu, i.e. either `///` or `/**  ....`
and make sure to document that this **appends** data to `ranks`, and does not clear it.
Thus maybe also rename the function to `computeAndAppendRankVector`